staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -407,6 +412,81 @@ func (obj InfoObject) Name() string {
 	)
 }
 
+func (obj InfoObject) GroupVersionKind() schema.GroupVersionKind {
+	return obj.Info.Mapping.GroupVersionKind
+}
+
+// mask conceals any secret values and returns the masked from/to versions
+// of the object.
+//
+// All secret values in the objects will be masked with a fixed-length
+// asterisk mask. If two values are different, an additional suffix will
+// be added so they can be diffed.
+func mask(from, to runtime.Object) (runtime.Object, runtime.Object, error) {
```

Comment:
This is definitely looking better, but I'm sure it can still be improved a little bit. The "three return params" of unstructuredMap isn't great. Anything we can do to improve it further?

Reply:
@apelisse I don't have any great ideas how to avoid it since it needs access to both from/to unstructs, and returning one of them means that we have to call this twice unnecessarily. Perhaps, I was thinking, maybe we can store the unstructs from/to in a struct (maybe something like a `Masker` or something) and then make this a method receiver of the `Masker` struct. Is this ok or you have something else in mind?